qiskit/circuit/library/n_local/evolved_operator_ansatz.py

```diff
@@ -93,7 +93,10 @@ def evolved_operator_ansatz(
     if reps < 0:
         raise ValueError("reps must be a non-negative integer.")
 
-    if isinstance(operators, BaseOperator):
+    # Check for SparseObservable first (it's iterable, so isinstance check must come before len check)
+    if isinstance(operators, SparseObservable):
+        operators = [operators]
+    elif isinstance(operators, BaseOperator):
```

Comment:
You can just merge these instance checks, as
```suggestion
    if isinstance(operators, (BaseOperator, SparseObservable)):
```